x-pack/plugins/index_management/public/store/selectors/index.js

```diff
@@ -26,19 +27,30 @@ export const getFilteredIds = (state) => state.indices.filteredIds;
 export const getRowStatuses = (state) => state.rowStatus;
 export const getTableState = (state) => state.tableState;
 
-
 export const getIndexStatusByIndexName = (state, indexName) => {
   const indices = getIndices(state);
   const { status } = indices[indexName] || {};
   return status;
 };
 const defaultFilterFields = ['name', 'uuid'];
+
+const filterByToggles = (indices, toggles) => {
```

Comment:
I was having a hard time following this because it wasn't apparent what the key names or values of `toggles` represented, or how they related to the index properties. I also may be misreading this logic, but it looks like we will only show an index if _all_ applicable toggles are visible, but I think it's a better UX if _any_ of the applicable toggles are visible.

What do you think of rewriting this to be a bit more verbose? This suggestion hinges on some of my other suggestions.

```js
const filterByToggles = (indices, toggleNameToVisibleMap) => {
  const togglesByName = getToggleExtensions().reduce((byName, toggle) => ({
    ...byName,
    [toggle.name]: toggle,
  }), {});

  const toggleNames = Object.keys(togglesByName);

  // An index is visible if ANY applicable toggle is visible.
  return indices.filter((index) => {
    return toggleNames.some(toggleName => {
      if (!togglesByName[toggleName].matchIndex(index)) {
        return true;
      }

      const isVisible = (
        toggleNameToVisibleMap[toggleName] === true
        || toggleNameToVisibleMap[toggleName] === undefined
      );

      return isVisible;
    );
  });
};
```

Reply:
done